src/domain/locking/entities/schemas/rank.schema.ts

```diff
@@ -1,25 +1,7 @@
-import { Rank } from '@/domain/locking/entities/rank.entity';
-import { JSONSchemaType, Schema } from 'ajv';
-import { buildPageSchema } from '@/domain/entities/schemas/page.schema.factory';
+import { z } from 'zod';
+import { buildZodPageSchema } from '@/domain/entities/schemas/page.schema.factory';
 
-export const RANK_SCHEMA_ID =
-  'https://safe-client.safe.global/schemas/locking/rank.json';
+// TODO: Add and test once defined
+export const RankSchema = z.object({});
 
-export const rankSchema: JSONSchemaType<Rank> = {
-  $id: RANK_SCHEMA_ID,
-  type: 'object',
-  properties: {
-    address: { type: 'string' },
-    rank: { type: 'string' },
-    lockedAmount: { type: 'string' },
-  },
-  required: ['address', 'rank', 'lockedAmount'],
-};
-
-export const RANK_PAGE_SCHEMA_ID =
-  'https://safe-client.safe.global/schemas/locking/rank-page.json';
-
-export const rankPageSchema: Schema = buildPageSchema(
-  RANK_PAGE_SCHEMA_ID,
-  rankSchema,
-);
+export const RankPageSchema = buildZodPageSchema(RankSchema);
```

Comment:
I wonder if we should declare the different `TypePageSchema` as it is basically just setting an actual type for a `Page`. We can see how this evolves but one alternative would be that we would not have a const/type to represent each paginated response 🤔

Reply:
I am impartial. Whilst migrating other schemas, I noticed that there are often no "page schemas" and `results` are mapped and validated directly.

I am in favour of seeing how this develops and aligning them all once we've fully migrated to `zod`. What do you think?

Reply:
> I am in favour of seeing how this develops and aligning them all once we've fully migrated to `zod`. What do you think?

Let's see how it develops then 👍